pkg-manager/plugin-commands-installation/src/install.ts

```diff
@@ -329,6 +329,15 @@ export type InstallCommandOptions = Pick<Config,
   confirmModulesPurge?: boolean
 } & Partial<Pick<Config, 'modulesCacheMaxAge' | 'pnpmHomeDir' | 'preferWorkspacePackages' | 'useLockfile'>>
 
+export function shouldFreezeLockfileIfExists (opts: InstallCommandOptions, onCI: boolean): boolean {
+  if (opts.frozenLockfileIfExists != null) {
+    return opts.frozenLockfileIfExists
+  }
+  return onCI && !opts.lockfileOnly &&
+    opts.rawLocalConfig['frozen-lockfile'] !== false &&
+    opts.rawLocalConfig['prefer-frozen-lockfile'] !== false
```

Comment:
when `prefer-frozen-lockfile` is set to anything other than `true` or `false`, it will be treated as the default value which is `true`.
`frozen-lockfile` comes from cli args, and it should be parsed as boolean?